### PR TITLE
Feat/login backend

### PR DESCRIPTION
**Pull request by Bighero0122**
- add authrouter
- modify user entity
- add login controller
- add utils/jwt
- add user service

### CODE REVIEW

**Comment by Bighero0122** [comment]
Do i have to merge pull request?

File: server/consts/index.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
 import ERRORS from "./errors";
 import * as MESSAGES from "./message";
-
 export { MESSAGES, ERRORS };
+export * as REASON_CODE from "./reasonCode";
```
Comment (genie099) [inline server/consts/index.ts]: This one should be in the line 3 and need to be exported as the part of above object.

File: server/controllers/auth/logIn.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { body } from "express-validator";
+import httpStatus from "http-status";
+
+import { CustomError } from "errors";
+
+import { userService } from "services";
+import { Logger, errorHandlerWrapper } from "utils";
+import { REASON_CODE } from "consts";
+import { comparePassword } from "utils/password";
+import { jwtSign } from "utils/jwt";
```
Comment (genie099) [inline server/controllers/auth/logIn.ts]: Please sort this imports

File: server/entities/User.entity.ts [hunk 3]
```diff
@@ -6,27 +6,18 @@ import { CoreEntity } from "./Core.entity";
...
 
-  @Column({ name: "display_name", nullable: true })
-  displayName?: string;
-
-  @Column({ name: "avatar", nullable: true })
-  avatar?: string;
-
-  @Column({ name: "notification_setting", length: 3 })
-  notificationSetting: string;
-
```
Comment (genie099) [inline server/entities/User.entity.ts]: This entity is different from DB diagram

File: server/routes/auth.router.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { authController } from "controllers";
+import express from "express";
+
+const authRouter = express.Router();
+
+// User Log In
+authRouter.post(
+  "log-in",
```
Comment (genie099) [inline server/routes/auth.router.ts]: url is incorrect. this one should be `/login`

File: server/routes/index.ts [hunk 5]
```diff
@@ -1,9 +1,11 @@
 import express from "express";
 
 import testRouter from "./test.route";
+import authRouter from "./auth.router";
 
 const appRoutes = express.Router();
 
 appRoutes.use("/test", testRouter);
+appRoutes.use("./auth", authRouter);
```
Comment (genie099) [inline server/routes/index.ts]: should remove `.`

File: server/routes/auth.router.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  authController.logInValidator(),
+  authController.logIn
```
Comment (genie099) [inline server/routes/auth.router.ts]: `logIn => login`

**Review by genie099 — COMMENTED** [review]
@Bighero0122 I left comment. Please have a look and solve it asap.

**Review by genie099 — APPROVED** [review]

**Pull request merged**